### PR TITLE
[IMP] core, requirements: bump minimal python version to 3.10

### PR DESCRIPTION
**Pull request by d-fence**
Now that the Debian 12 ("Bookworm") is out with Python 3.11 as the default, it's time to update our requirements.

Reminder of the constraints for our requirements:

We try choose the smallest version from the Ubuntu/Debian corresponding package (python3-...).

Also, if we find that one of the package was patched by the Debian/Ubuntu maintainer, we choose the version from which the patch is coming.

So, before this commit, the version were choose between Debian 11 and Ubuntu 22.04. With this commit, we can simplify the requirements because of a better matching between "Jammy" and "Bookworm".

About the choice of the python version:

* Ubuntu 22.04 ("Jammy") provides 3.10
* Debian 12 ("Bookworm") provides 3.11
* Some features that only exists in 3.9 will be needed in a near future
* 3.9 is a small release

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,58 +4,52 @@ Babel==2.9.1  # min version = 2.6.0 (Focal with security backports)
...
 geoip2==2.9.0
-gevent==20.9.0 ; python_version <= '3.9'
 gevent==21.8.0 ; python_version == '3.10'  # (Jammy)
 gevent==22.10.2; python_version > '3.10'
-greenlet==0.4.17 ; python_version <= '3.9'
 greenlet==1.1.2 ; python_version == '3.10'  # (Jammy)
 greenlet==2.0.2 ; python_version > '3.10'
-idna==2.10
-Jinja2==2.11.3 ; python_version <= '3.10'  # min version = 2.10.1 (Focal - with security backports)
+idna==2.10  # requests 2.25.1 depends on idna<3 and >=2.5
```
Comment (gdgellatly) [inline requirements.txt:15]: ? The comment seems incongruent with the requirement?

**Comment by Whenrow** [comment]
I would adapt the doc too 

https://github.com/odoo/documentation/blob/90c86e6462a6d87705748409a0bd5b3812728a81/content/administration/install/source.rst#L121

**Comment by tde-banana-odoo** [comment]
I am strongly against this change !

Otherwise you would be lost and could maybe be worried I am giving up raling for principe.

**Comment by Xavier-Do** [comment]
> I am strongly against this change !
> 
> Otherwise you would be lost and could maybe be worried I am giving up raling for principe.

I would be really lost if you were the one r+ing this pr.

**Comment by tde-banana-odoo** [comment]
@robodoo r+

PAN, don't challenge me !

In case of doubt, I am away. Anyway there are enough people that can r- .

**Comment by Xavier-Do** [comment]
Fortunately the documentation is not r+. So lets leave it like that

**Comment by amigrave** [comment]
> > I am strongly against this change !
> > Otherwise you would be lost and could maybe be worried I am giving up raling for principe.
> 
> I would be really lost if you were the one r+ing this pr.

Is that reverse-reverse-reverse psychology ?

**Comment by d-fence** [comment]
robodoo rebase-ff

**Pull request closed without merging**